src/backend/dx11/src/device.rs

```diff
@@ -1004,31 +1123,71 @@ impl hal::Device<Backend> for Device {
 
     fn create_descriptor_pool<I>(
         &self,
-        _max_sets: usize,
-        _descriptor_pools: I,
+        max_sets: usize,
+        ranges: I,
     ) -> DescriptorPool
     where
         I: IntoIterator,
         I::Item: Borrow<pso::DescriptorRangeDesc>
     {
-        // TODO: descriptor pool
+        let count = ranges.into_iter().map(|r| r.borrow().count).sum::<usize>() * max_sets;
```

Comment:
I don't think we need to multiply that total count in ranges with the max sets. Those are independent limits.

Reply:
Maybe my understanding of descriptor pools are wrong. You supply arbitrary number of descriptor types and lengths which it hands out for allocation. Is the amount of sets you provide to pool creation not related at all to how many descriptors you can allocate?

Reply:
I'm open to consider a different interpretation, but I see it as a different limit. E.g. you can create up to as many of each of the descriptors AND up to this many of descriptor sets in total. How exactly each of the sets is using the descriptors is not important.
Let me invite @msiglreith to clarify, if they have a clearer understanding.

Reply:
@kvark is correct ([Slide 22](http://32ipi028l5q82yhj72224m8j-wpengine.netdna-ssl.com/wp-content/uploads/2016/05/Most-common-mistakes-in-Vulkan-apps.pdf) for IHV response on this)